python/cudf/cudf/core/frame.py

```diff
@@ -1580,7 +1580,13 @@ def _get_sorted_inds(self, by=None, ascending=True, na_position="last"):
         if np.isscalar(ascending):
             ascending = [ascending] * len(to_sort)
 
-        return libcudf.sort.order_by(to_sort, ascending, na_position)
+        # Only use a stable sort in pandas-compat mode
```

Comment:
I’d like to propose using stable sorts always. If I recall correctly there is no real performance difference in Thrust for these two calls. If there is no performance loss, I would prefer to ensure that we always match pandas regardless of compatibility mode. The compatibility mode flag is best used when it incurs a significant loss of performance in order to establish identical behavior. For cases with similar perf or only a small loss of perf, I would always match pandas. It’s one fewer thing for our users to worry about.